File: src/facade/python/DataBaseFacade.py
from src.decorator.facade.python.db.Statement import Statement
from src.decorator.facade.python.db.Transaction import Transaction
from src.decorator.facade.python.db.Connection import Connection


class DataBaseFacade:
    def execute(self, sql: str):
        statement = Statement(sql)
        connection = Connection()
        transaction = Transaction()

        connection.connect("*dbUrl*")

        transaction.begin_transaction()
        connection.execute_statement(statement)
        print(connection.get_result_set().get_result())
        transaction.end_transaction()

File: src/facade/java/com/my/DataBaseFacade.java
package src.facade.java.com.my;

import src.facade.java.com.my.db.Connection;
import src.facade.java.com.my.db.Statement;
import src.facade.java.com.my.db.Transaction;

public class DataBaseFacade {
    public void execute(String sql) {
        Statement statement = new Statement(sql);
        Connection connection = new Connection();
        Transaction transaction = new Transaction();

        connection.connect("*dbUrl*");

        transaction.beginTransaction();
        connection.executeStatement(statement);
        System.out.println(connection.getResultSet().getResult());
        transaction.endTransaction();
    }
}
